mesonbuild/modules/python.py

```diff
@@ -144,15 +144,13 @@ def __init__(self, python: 'PythonExternalProgram', interpreter: 'Interpreter'):
         })
 
     @permittedKwargs(mod_kwargs)
-    @typed_kwargs('python.extension_module', *_MOD_KWARGS, allow_unknown=True)
+    @typed_kwargs('python.extension_module', *_MOD_KWARGS, _DEFAULTABLE_SUBDIR_KW, allow_unknown=True)
     def extension_module_method(self, args: T.List['TYPE_var'], kwargs: ExtensionModuleKw) -> 'SharedModule':
         if 'install_dir' in kwargs:
-            if 'subdir' in kwargs:
+            if kwargs['subdir'] is not None:
                 raise InvalidArguments('"subdir" and "install_dir" are mutually exclusive')
         else:
-            subdir = kwargs.pop('subdir', '')
-            if not isinstance(subdir, str):
-                raise InvalidArguments('"subdir" argument must be a string.')
+            subdir = kwargs['subdir'] or ''
```

Comment:
I can put the `pop` back for now, but `TypedDict` doesn't supported `.pop()` (at least ATM, maybe with the new `NorRequired` annotation it will?), so we ultimately have to drop it to make mypy happy.

Reply:
but also that means we have to write it as:
```python
subdir = kwargs.pop('subdir') or ''
```
which look weird and will absolutely need a comment along the lines of `this will either be a str or None, so don't use `pop('subdir', '')` here, that won't work`

Reply:
Well look at that, `NotRequired` does allow this to work. `pop()` has been put back